src/services/resource_loader/resource_loader.ts

```diff
@@ -0,0 +1,67 @@
+import { Interfaces, Resource, Service } from "../../../mod.ts";
+import { walk } from "./deps.ts";
+
+interface IOptions {
+  paths_to_resources: string[];
+}
+
+export class ResourceLoaderService extends Service {
+  #options: IOptions;
+
+  constructor(options: IOptions) {
+    super();
+    this.#options = options;
+  }
+
+  public async runAtStartup(
+    options: Interfaces.IServiceStartupOptions,
+  ): Promise<void> {
+    for (const i in this.#options.paths_to_resources) {
+      const basePath = this.#options.paths_to_resources[i];
```

Comment:
Is there a reason for not having `for (const basePath of this.#options.paths_to_resources)` ?

Reply:
cause i keep forgetting about that lmfao. i added it here: [4faa89b](https://github.com/drashland/drash/pull/644/commits/4faa89b66ad65750cf9ec97b861e3fc637106ab0)

also, i changed the `forEach` loop to all be `for` and changed from `walk` to `walkSync`. couldn't figure out why leaking async ops were happening, but basically the resource loader would work sometimes (it would load all resources or load just one resource. i even added a deferred call to `await` for the resource loader to finish and it made no difference)